skate-plugin/project-gen/src/jvmMain/kotlin/slack/tooling/aibot/ChatPresenter.kt

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2024 Slack Technologies, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package slack.tooling.aibot
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import com.slack.circuit.runtime.presenter.Presenter
+
+class ChatPresenter : Presenter<ChatScreen.State> {
+  @Composable
+  override fun present(): ChatScreen.State {
+    var messages by remember { mutableStateOf(emptyList<Message>()) }
+    var currentInput by remember { mutableStateOf("") }
+
+    return ChatScreen.State(messages = messages, currentInput = currentInput) { event ->
+      when (event) {
+        is ChatScreen.Event.SendMessage -> {
+          val newMessage = Message(event.message, isMe = true)
+          messages = messages + newMessage
+          val response = Message(callApi(event.message), isMe = false)
+          messages = messages + response
+          currentInput = "" // clear chat after sending
```

Comment:
I would suggest keeping "currentInput" entirely in the UI, and just use this event to signal a new sent message with the text received from the UI. What do you think?

Reply:
Yeah this makes sense, I should just keep it in UI. Will make this change! Thanks

Reply:
I removed it in f27ddc1af95d1c0b90e33622407588cd464def09 under `ChatPresenter.kt`